code/modules/reagents/chemistry/holder.dm

```diff
@@ -556,7 +559,8 @@ var/const/INJECT = 5 //injection
 			update_total()
 			if(!safety)//So it does not handle reactions when it need not to
 				handle_reactions()
-			my_atom.on_reagent_change()
+			if(my_atom)
```

Comment:
Yeah, I'm working on it.  Also probably gonna apply the opt memes I've picked up from Atmos and shit